src/Domain.LinnApps/Reports/ShortagesReportService.cs

```diff
@@ -0,0 +1,73 @@
+﻿namespace Linn.Purchasing.Domain.LinnApps.Reports
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+
+    using Linn.Common.Persistence;
+    using Linn.Common.Reporting.Models;
+    using Linn.Purchasing.Domain.LinnApps.Reports.Models;
+
+    public class ShortagesReportService : IShortagesReportService
+    {
+        private readonly IRepository<ShortagesEntry, string> shortagesEntryView;
+
+        public ShortagesReportService(
+            IRepository<ShortagesEntry, string> shortagesEntryView)
+        {
+            this.shortagesEntryView = shortagesEntryView;
+        }
+
+        public IEnumerable<ResultsModel> GetReport(
+            int purchaseLevel,
+            string vendorManager)
+        {
+            
+            var results = this.shortagesEntryView.FilterBy(x => (Convert.ToInt32(x.PurchaseLevel) <= purchaseLevel) &&
+                                                                            (vendorManager == "ALL" || vendorManager == x.VendorManagerCode)).ToList();
```

Comment:
how do you actually run this report for vendorManager = "ALL" ? I can't see how you could select 'ALL' from your options screen dropdown